web/src/user_card_popover.js

```diff
@@ -325,6 +326,26 @@ function render_user_card_popover(
         $popover_element[0],
         {
             placement: popover_placement,
+            popperOptions: {
+                modifiers: [
+                    {
+                        // Show popover at the middle of window on small widths.
+                        name: "offset",
+                        options: {
+                            offset: ({reference, _placement, popper}) => [
+                                0,
+                                // On small screens, we want to center the popover
+                                // so, we calculate the offset from the reference
+                                // element to the center of the window.
+                                window.innerWidth < media_breakpoints_num.md
+                                    ? -(reference.x + reference.width) +
+                                      (window.innerWidth - popper.width) / 2
+                                    : 0,
+                            ],
+                        },
+                    },
+                ],
+            },
```

Comment:
This more or less duplicates `get_props_for_popover_centering`, which also does a backdrop; can you deduplicate these two implementations?

Reply:
Oh! Looking at the code looks like user card popover was supposed to use this but it wasn't due to a bug.

I did a simple refactor which also fixed this.

Reply:
I refactored it to show popovers with `show_as_overlay_on_mobile: true`
as overlay on mobile widths instead of every popover checking for mobile width on their own.